flake.nix

```diff
@@ -42,107 +42,64 @@
+        ./build02
+        ./build03
+        ./build04
+      ];
 
-        perSystem = { config, pkgs, ... }: {
-          treefmt.imports = [ ./treefmt.nix ];
+      perSystem = { config, pkgs, ... }: {
+        treefmt.imports = [ ./treefmt.nix ];
 
-          packages.pages = pkgs.runCommand "pages"
-            {
-              buildInputs = [ pkgs.python3.pkgs.mkdocs-material ];
-            } ''
-            cp -r ${pkgs.lib.cleanSource ./.}/* .
-            mkdocs build --strict --site-dir $out
-          '';
+        packages.pages = pkgs.runCommand "pages"
+          {
+            buildInputs = [ pkgs.python3.pkgs.mkdocs-material ];
+          } ''
+          cp -r ${pkgs.lib.cleanSource ./.}/* .
+          mkdocs build --strict --site-dir $out
+        '';
 
-          hercules-ci.github-pages.settings.contents = config.packages.pages;
-        };
+        hercules-ci.github-pages.settings.contents = config.packages.pages;
+      };
 
-        flake.nixosConfigurations =
-          let
-            inherit (inputs.nixpkgs.lib) nixosSystem;
-            common = [
-              { _module.args.inputs = inputs; }
-              { srvos.flake = inputs.self; }
-              inputs.sops-nix.nixosModules.sops
-              inputs.srvos.nixosModules.server
-
-              inputs.srvos.nixosModules.mixins-telegraf
-              { networking.firewall.allowedTCPPorts = [ 9273 ]; }
-            ];
-          in
-          {
-            build01 = nixosSystem {
-              system = "x86_64-linux";
-              modules =
-                common
-                ++ [
-                  ./build01/configuration.nix
-                  inputs.srvos.nixosModules.hardware-hetzner-online-amd
-                ];
-            };
-
-            build02 = nixosSystem {
-              system = "x86_64-linux";
-              modules =
-                common
-                ++ [
-                  ./build02/configuration.nix
-                  inputs.srvos.nixosModules.mixins-nginx
-                  inputs.srvos.nixosModules.hardware-hetzner-online-amd
-                ];
-            };
-
-            build03 = nixosSystem {
-              system = "x86_64-linux";
-              modules =
-                common
-                ++ [
-                  ./build03/configuration.nix
-                  inputs.srvos.nixosModules.mixins-nginx
-                  inputs.srvos.nixosModules.hardware-hetzner-online-amd
-                ];
-            };
-
-            build04 = nixosSystem {
-              system = "aarch64-linux";
-              modules =
-                common
-                ++ [
-                  ./build04/configuration.nix
-                  inputs.disko.nixosModules.disko
-                ];
-            };
-          };
+      flake.lib.nixosSystem = args:
+        inputs.nixpkgs.lib.nixosSystem ({ specialArgs = { inherit inputs; }; } // args);
+
+      flake.nixosModules = {
+        common = ./roles/common.nix;
```

Comment:
What's this for?

Reply:
Hmm, this didn't belong in this PR. It's related to the deeper structure changes I started working on. See  https://github.com/nix-community/infra/pull/589